modules/core/plugin-container/src/main/java/org/rhq/core/pc/inventory/InventoryManager.java

```diff
@@ -1928,24 +1931,24 @@ private Configuration mergeExistingResourcePluginConfiguration(Resource resource
 
         // If there is no update necessary just return the current plugin config of the existing resource
         Configuration result = resource.getPluginConfiguration();
-
         ConfigurationDefinition configDef = resource.getResourceType().getPluginConfigurationDefinition();
         if (null == configDef) {
             return result;
         }
-
+        Map<String, PropertyDefinition> propDefs = configDef.getPropertyDefinitions();
         Configuration existingPluginConfig = resource.getPluginConfiguration().deepCopy(false);
-        Configuration defaultPluginConfig = ConfigurationUtility.createDefaultConfiguration(configDef);
+
         boolean configChanged = false;
 
         // for each property, update the existing plugin config if discovery has set a non-default value
         for (String propertyName : pluginConfig.getAllProperties().keySet()) {
             Property discoveredProp = pluginConfig.get(propertyName);
-            Property defaultProp = defaultPluginConfig.get(propertyName);
-            if (!discoveredProp.equals(defaultProp)) {
+            // Only overrides the property if it is a read-only property
+            // See BZ: 1379834
+            if (propDefs.get(propertyName).isReadOnly()) {
```

Comment:
Although the _override strategy_ idea is a good one, it isn't something that we should really be doing in the maintenance stream. Not only would it force a version bump in the plug-in XSD, it would require this new tag/attribute to be defined for many resources and their configuraitons.

I do not think there should be any harm in only applying the the change to read-only properties. We are already doing that. All we are saying now is that we do not want to replace user provided configuration.

However, are we sure that we also do not want to verify the value requires updating? For example:
```
-            if (!discoveredProp.equals(defaultProp)) {
+            // Only overrides the property if it is a read-only property
+            // See BZ: 1379834
+            if (propDefs.get(propertyName).isReadOnly() && !discoveredProp.equals(defaultProp)) {
```
Of course, I am still not sure what the purpose of `discoveredProp.equals(defaultProp)` is even for, considering that discovery should always discover a different property then the default.